src/Microsoft.AspNetCore.Sockets.Common/MessageBatchFormatter.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    public static class MessageBatchFormatter
+    {
+        private const byte BatchTerminator = (byte)';';
+        private const byte TextFormatIndicator = (byte)'T';
+        private const byte BinaryFormatIndicator = (byte)'B';
+
+        public static bool TryFormatMessages(IEnumerable<Message> messages, Span<byte> buffer, MessageFormat format, out int bytesWritten)
```

Comment:
Actually, moving the "format-discriminator" writing/parsing code out of this might make this much cleaner, since that's only ever needed at the very start of the entire stream. I'm going to refactor a bit to make this work.